include/boost/decimal/detail/cmath/ilogb.hpp

```diff
@@ -31,9 +31,13 @@ constexpr auto ilogb(T d) noexcept -> std::enable_if_t<detail::is_decimal_floati
         return FP_ILOGBNAN;
     }
 
-    const auto offset { detail::num_digits(d.full_significand()) - 1 };
+    const auto offset =
+        static_cast<int>
+        (
+            detail::num_digits(d.full_significand()) - static_cast<int>(INT8_C(1))
```

Comment:
This double cast adds new warning for redundant casting. `num_digits` returns an int and 1 is an int